lib/cli.ex

```diff
@@ -1,7 +1,20 @@
 defmodule Onigumo.CLI do
-  def main([component]) do
-    module = Module.safe_concat("Onigumo", component)
+  def main(argv) do
+    {_parsed, args, _invalid} = OptionParser.parse(argv, strict: [])
+
+    module =
+      args
+      |> get_component
+      |> safe_concat
+
     root_path = File.cwd!()
     module.main(root_path)
   end
+
+  defp get_component([]), do: "Downloader"
+  defp get_component([component | _]), do: component
+
+  defp safe_concat(component) do
+    Module.safe_concat("Onigumo", component)
+  end
```

Comment:
As mentioned above. Thanks to abandoning _safe_concat_ as unsafe 😃, exposing new errors and abusing reflection, we won’t need this function altogether.

```suggestion
```